backend/src/main/resources/doc/openapi.json

```diff
@@ -1191,6 +1194,34 @@
                 "required": [
                     "type"
                 ]
+            },
+            "Badge": {
+                "type": "object",
+                "title": "Badge",
+                "description": "Badge owned by the user",
+                "properties": {
+                    "type": {
+                        "type": "string"
```

Comment:
Type. Do you mean the check-up type, don't you? I would use the already existing enum definition here. I'm not on computer so I don't know the name exactly but it's used for example in examinations endpoints.

Reply:
I meant type of the badge itself, for example `OPASEK` or `CEPICE` etc

Reply:
Understood. What about a new enum definition in the OpenApi to share the enum values with the FE. I suppose that it will be a constant list.

Reply:
Yes, I like it, then our enum would the source of truth

Reply:
Be careful. The number of badges may grow (and it will). We need to be backward compatible with older apps. Enums are the worst thing to keep compatible.

Reply:
Maybe it would be better due to these reasons to keep it as a separate endpoint? I know a very silly endpoint. Just think about it, it is not necessary.

Reply:
Since no badge types are sent from FE to BE, I think any enum changes wouldn't break anything, but yes, when the change comes to the enum our FE would have to make the change on their side, they would would have to it anyways, I'd say, since it doesn't matter where we make a change in the badges memory/db/enum, they still would have it react on it

Reply:
I don't have a strong opinion on it, but the more we discuss it, the more I think that having a separate endpoint for badges would be the best idea, because it would allow easier extensibility in the future, i.e.

Keep as is (in terms of parameters and headers):
`/confirm` - adds badge and points automatically, without anything special required from FE

Adding new APIs:
(GET) `/badges/{userId}/` - returns all badges for user
(GET) `/badges`                - returns all available badges in general

Badge type wouldn't be returned as enum to avoid breaking changes

@martin-ladecky @rposkocil WDYT?

Reply:
Can we also invite someone from FE into this discussion?

Reply:
@MartinMacek, @mzdm - any opinion?

Reply:
I also don't have a strong opinion on it. I don't know much about the planned badge system but if it would be with enum we could already use some of this in the onboarding form in the reward screens where authentication is not available yet. We are hardcoding it right now. But I don't know how stable the badge system currently is and how many changes are expected. So really hard to say. WDYT @MartinMacek

Reply:
In case we are planning to extend the badge collection in the future, it makes a lot of sense to me, to have the badge collection on separate endpoint - including svg assets. 

Worst case scenario is a new enum from api but missing asset in old app version. (This happens on android quiet often - some people don’t update apps)

Best case scenario is that we cache the badge collection in app and sync it on app start in the background. 
Further optimisations might follow to save some traffic (like collection version in request param...).

Definitely not easiest solution, but IMO robust one. 

TBH, I too do not know very much about the badge system yet. Open to discussion

Reply:
Come on, guys! Nedelejte mi v tom bramboracku. :-D A napisu to cesky, at se neco neztrati v prekladu. :-)
Pojdme se spolecne dohodnout na nejakem paradigmatu. Resim tohle tema v podstate s kazdym PR a navrhem API.

Domluvili jsme se kdysi, ze pravda bude v dokumentaci a ze chceme razit proces definition first - tedy pravda je v OpenApi jsonu. Proto jsou vsechny enumy jako sex a examinationType (ktery take nemusi byt nemenny) definovane v JSONu.

Dalsi moje pripominka k roli BE. Jak ho do ted chapal jako data providera a drzitele pravdy. Tedy, ze poskytuje pro MA informace, ktere jsou ve forme definovane v OpenApi. Posilani nejake definice z BE na endpointu je zavadeni uplne jineho pristupu.

Take vubec nerozumim tomu, proc by BE mel resit assety. Mozna mi neco unika. Jestli je to kvuli kompatibilite, tak by me zajimalo jakym zpusobem budeme vydavat nove verze a aktualizovat server a aplikace. Melo by to jit ruku v ruce a za me nejlepe, kdyby MA kontrolovala dostupnost nove verze, tak jak je tomu bezne.

Jakmile se shodneme, muzeme to sjednotit a hlavne mit v tom jasno. Chtel bych aby tyhle arch. veci byli definovane a jasne sepsane. Potrebuju vedet, co se od BE ceka. :-)

Reply:
Yes, let's not overengineer, and go with enum. If something goes wrong, blame the design team :D

K těm assetům, to je taky otázka no. Jestli máme brát assety podle `{TYPE}` a `{LEVEL}` lokálně z appky z assetů takže `/assets/badges/{TYPE}_{LEVEL}` tak to může prvních pár měsíců fungovat ale musí se fakt dávat pozor aby ty levely seděly s tím co je lokálně v assetech..

Reply:
Awesome, I'll rewrite it to `Enum`

Reply:
Done in 7252797

Reply:
Enum is ok.
K assetům: Za mňa by si appka mala sama nájsť obrázok na základe typu enumu. Prípadne si to môže stiahnuť z firebase, ... . Ideálne ale ako obrázok v appke a na základe stringu z BE si appka sama vyberie. Nerád by som, aby server posielal obrázky. To nie je dobrý design.